etl/steps/data/garden/energy/2022-12-13/owid_energy.py

```diff
@@ -0,0 +1,215 @@
+    assert len(columns_with_inf) == 0, f"Infinity values detected in columns: {columns_with_inf}"
+
+    # Set index and sort conveniently.
+    tb_combined = tb_combined.set_index(["country", "year"], verify_integrity=True).sort_index()
+
+    return cast(catalog.Table, tb_combined)
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load data.
+    #
+    # Read all required datasets.
+    ds_energy_mix = catalog.Dataset(ENERGY_MIX_DATASET_PATH)
+    ds_fossil_fuels = catalog.Dataset(FOSSIL_FUEL_PRODUCTION_DATASET_PATH)
+    ds_primary_energy = catalog.Dataset(PRIMARY_ENERGY_CONSUMPTION_DATASET_PATH)
+    ds_electricity_mix = catalog.Dataset(ELECTRICITY_MIX_DATASET_PATH)
+
+    # Gather all required tables from all datasets.
+    tb_energy_mix = ds_energy_mix[ENERGY_MIX_TABLE_NAME].reset_index()
+    tb_fossil_fuels = ds_fossil_fuels[FOSSIL_FUEL_PRODUCTION_TABLE_NAME].reset_index()
+    tb_primary_energy = ds_primary_energy[PRIMARY_ENERGY_CONSUMPTION_TABLE_NAME].reset_index()
+    tb_electricity_mix = ds_electricity_mix[ELECTRICITY_MIX_TABLE_NAME].reset_index()
+
+    # Load countries-regions dataset (required to get ISO codes).
+    countries_regions = catalog.Dataset(COUNTRIES_REGIONS_DATASET_PATH)["countries_regions"].reset_index()[
+        ["name", "iso_alpha3"]
+    ]
+
+    # Population data will also be loaded (used only to add a population column, and not to create any other derived
+    # variables). Historical regions will be added to the population.
+
+    # Load gdp (used only to add gdp column, and no other derived variables).
+    gdp = (
+        pd.DataFrame(catalog.Dataset(GDP_DATASET_PATH)["maddison_gdp"])
+        .reset_index()[["country", "year", "gdp"]]
+        .dropna()
+    )
+
+    # Load mapping from variable names in the component dataset to the final variable name in the output dataset.
+    variable_mapping = pd.read_csv(VARIABLE_MAPPING_FILE).set_index(["source_variable"])
+
+    #
+    # Process data.
+    #
+    # Combine all tables.
+    tables = {
+        "energy_mix": tb_energy_mix.drop(columns=["country_code"], errors="ignore"),
+        "fossil_fuel_production": tb_fossil_fuels,
+        "primary_energy_consumption": tb_primary_energy.drop(columns=["gdp", "population", "source"], errors="ignore"),
+        "electricity_mix": tb_electricity_mix.drop(
+            columns=["population", "primary_energy_consumption__twh"], errors="ignore"
+        ),
+    }
+    tb_combined = combine_tables_data_and_metadata(
+        tables=tables,
+        countries_regions=countries_regions,
+        gdp=gdp,
+        variable_mapping=variable_mapping,
+    )
+
+    #
+    # Save outputs.
+    #
+    ds_garden = catalog.Dataset.create_empty(dest_dir)
+    # Gather metadata sources from all tables' original dataset sources.
+    ds_garden.metadata.sources = gather_sources_from_tables(tables=list(tables.values()))
+    # Get the rest of the metadata from the yaml file.
+    ds_garden.metadata.update_from_yaml(METADATA_PATH)
```

Comment:
You could use the new method: `ds_garden.update_metadata(METADATA_PATH)`. This also affects the building of table `tb_combined`.

Note that you should call this function after adding the tables to the dataset.